interface-impl-amf/src/main/java/org/mule/amf/impl/model/MimeTypeImpl.java

```diff
@@ -173,22 +179,23 @@ public Object getInstance() {
   public List<ApiValidationResult> validate(String payload) {
     String mimeType = getMimeTypeForValue(payload);
 
-    PayloadValidator payloadValidator = payloadValidatorMap.computeIfAbsent(mimeType,
-                                                                            payloadMimeType -> getPayloadValidator(payloadMimeType)
-                                                                                .orElse(null));
+    AMFShapePayloadValidator payloadValidator = payloadValidatorMap.computeIfAbsent(mimeType,
+                                                                                    payloadMimeType -> getPayloadValidator(payloadMimeType));
 
     if (payloadValidator != null) {
-      return mapToValidationResult(payloadValidator.syncValidate(mimeType, payload));
+      return mapToValidationResult(payloadValidator.syncValidate(payload));
     }
     return of(
               new ExceptionApiValidationResult(new RuntimeException(format("Validator not found for %s", mimeType))));
   }
 
-  private Optional<PayloadValidator> getPayloadValidator(String mediaType) {
-    return ((ValidatorAware) shape).payloadValidator(mediaType);
+  private AMFShapePayloadValidator getPayloadValidator(String mediaType) {
+    return APIConfiguration.fromSpec(spec).withShapePayloadPlugin(new XmlValidationPlugin()).elementClient()
+        .payloadValidatorFor(shape, mediaType,
+                             ValidationMode.StrictValidationMode());
```

Comment:
Payload Validator needs to be extracted from a configuration based on the spec (RAML,OAS)

Reply:
Also XmlValidation plugin neeeds to be registered